src/test-run/index.ts

```diff
@@ -545,7 +545,8 @@ export default class TestRun extends AsyncEventEmitter {
     }
 
     private async _initRequestHooks (): Promise<void> {
-        await Promise.all(this.test.requestHooks.map(hook => this._initRequestHook(hook)));
+        if (!this.test.skip)
```

Comment:
The problem is that using this method outside doesn't imply that it may be skipped because of the test, there is nothing about it in the name. So we have two ways, add this information to the name or just move this condition outside in the upper method. I suggest the second way, also I think we can skip `_clearCookiesAndStorages`